examples/custom-node/src/evm/alloy.rs

```diff
@@ -43,22 +38,20 @@ where
     type Inspector = I;
 
     fn block(&self) -> &BlockEnv {
-        &self.inner.ctx_ref().block
+        Evm::block(&self.inner)
     }
 
     fn chain_id(&self) -> u64 {
-        self.inner.ctx_ref().cfg.chain_id
+        Evm::chain_id(&self.inner)
     }
 
     fn transact_raw(
         &mut self,
         tx: Self::Tx,
     ) -> Result<ResultAndState<Self::HaltReason>, Self::Error> {
-        if self.inspect {
-            self.inner.set_tx(tx.0);
-            self.inner.inspect_replay()
-        } else {
-            self.inner.transact(tx.0)
+        match tx {
+            CustomEvmTransaction::Op(tx) => Evm::transact_raw(&mut self.inner, tx),
+            CustomEvmTransaction::Payment(..) => todo!(),
```

Comment:
great this would be the where we'd need to execute this natively, we can keep this as todo! for now